### PR TITLE
DualStack kubernetes based IPv6 testing for Cilium

### PR DESCRIPTION
**Pull request by fristonio**
Update K8sServiceTests to include IPv6 tests for Kubernetes running in Dual Stack mode.

### CODE REVIEW

**Comment by fristonio** [comment]
test-me-please

File: test/helpers/utils.go
Comment (aanm) [inline test/helpers/utils.go]: Please invert the logic, i.e., set `case "1.12","1.13","1.14","1.15","1.16","1.17":` so that we don't need to add a new version each time a k8s verion is released.

File: test/k8sT/manifests/1.20/demo_ds_dualstack.yaml
Comment (aanm) [inline test/k8sT/manifests/1.20/demo_ds_dualstack.yaml]: Consider moving these files from `test/k8sT/manifests/1.20` to `test/k8sT/manifests`. It's likely that these files will work in newer k8s versions and we won't need to add a new directory each time a new k8s version is released.

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: This looks odd, we can't "know" the order the variable will be initialized so it would be better executing `helpers.DualStackSupported()` directly here.

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: just search and replace `dualStackSupportEnabled` with `helpers.DualStackSupported()` :sweat_smile:

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: Is this correct? Why are we adding `global` here?

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: Is this correct? Why are we adding `global` here?

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: Is this correct? Why are we adding `global` here?

File: test/k8sT/Services.go
Comment (aanm) [inline test/k8sT/Services.go]: Is this correct? Why are we adding `global` here?

**Review by aanm — CHANGES_REQUESTED** [review]

File: test/k8sT/Services.go
Comment (fristonio) [reply test/k8sT/Services.go]: Yeah, this is something I missed when rebasing the PR. The global prefix should not be here, I will remove these.
The test passes because we have backward compatibility for these helm variables in our CI.

File: test/helpers/utils.go
Comment (fristonio) [reply test/helpers/utils.go]: Yeah, makes sense.

File: test/k8sT/manifests/1.20/demo_ds_dualstack.yaml
Comment (fristonio) [reply test/k8sT/manifests/1.20/demo_ds_dualstack.yaml]: Cool, will do that.

**Review by aanm — APPROVED** [review]

**Comment by fristonio** [comment]
test-me-please

**Comment by fristonio** [comment]
retest-gke

**Comment by fristonio** [comment]
retest-runtime

File: test/provision/k8s_install.sh
Comment (christarazi) [inline test/provision/k8s_install.sh]:
```suggestion
# Configure default IPv6 route without this connectivity from host to
```

File: test/k8sT/Services.go
Comment (christarazi) [inline test/k8sT/Services.go]: Nit:

```
			node1IP, node2IP      string
			localSvc, k8s2LocalSvc string
```

**Review by christarazi — APPROVED** [review]
LGTM, minor nits. Doesn't require a second review from me.

File: test/k8sT/manifests/demo_ds_dualstack.yaml
Comment (sayboras) [inline test/k8sT/manifests/demo_ds_dualstack.yaml:7]: The file name might be a little bit confused if anyone didn't dig deeper for context (e.g. the comment in 1.19/demo_dualstack.yaml file). Basically, the manifest is _only_ for ipv6 service, but not for _dual stack_.

In my opinion, these files could be named as `*_ipv6.yaml` 🤔

**Review by sayboras — APPROVED** [review]
LGTM, one minor on file naming, happy to go either way 💯

File: test/k8sT/manifests/demo_ds_dualstack.yaml
Comment (fristonio) [reply test/k8sT/manifests/demo_ds_dualstack.yaml:7]: Good point. While for now the manifests only contain IPv6 services, going forward we might add some tests which are not IPv6 specific but rather target specific dual-stack features. From the top of my head, one such feature I can think of is to validate if Cilium honors the` ipFamilyPolicy` field(I will be working on this next). 
Happy to change the name of the files for now though. Thoughts @christarazi  @aanm ?

Comment (aanm) [reply test/k8sT/manifests/demo_ds_dualstack.yaml:7]: It's fine, the overall intent of the filename is to immediately tell to what it relates to.

**Comment by fristonio** [comment]
test-me-please

**Comment by fristonio** [comment]
retest-4.9

**Pull request merged**
